### ISSUE TITLE
exec shell script withlinux pipeline command always block forever

### ISSUE THREAD

**Issue by gao634209276** [azkaban/azkaban#1754]

**Comment by gao634209276** [comment]
my azkaban version: 3.43.0
I edit a command type job like this:
`
type=command
command=sh test.sh
`
and the test.sh write :
`
#!/bin/bash env
source ~/.bashrc
head my_file
head my_file | wc -l
count1=$(more my_file | wc -l)
echo ${count1}
`
when azkaban run this job, it always block in linux pipeline command,

**Comment by reallocf** [comment]
Hey @gao634209276 - isn't this the designed behavior of the `more` command?

From the `more` man page:
`More is a filter for paging through text one screenful at a time.`

When Azkaban simply launches a subshell that executes whatever commands are specified. I imagine it's being `blocked` because there's no user to actually page through `my_file`.

Can't you accomplish a similar objective with cat instead?
`count1=$(cat my_file | wc -l)`

Because cat doesn't require user input (which subshells launched by Azkaban aren't able to receive) it prints everything to the screen immediately (or in this case, to the pipe which then counts all the lines).

Does that address your question?

**Comment by gao634209276** [comment]
oh yeah, My example is not very good. If the text exceeds one page, `more` command will be blocked.
 I mean that command `count1=$(more my_file | wc -l)`,when I run a subshell use command line or set  crontab, it work good, but in azkaban it does work.  Where are the differences between crontab and azkaban on that case?

**Comment by gao634209276** [comment]
on this case, that one problem, There is another problem with this:
 shell script run `count1=$(more my_file | wc -l)` before, 
the next command for example is `curl url` to do somthing.
Due to azkaban run `count1=$(more my_file | wc -l)` being blocked, I kill the Job.
but I fount the next few commands run when I kill the job.
How did that happen？

**Comment by vasuonweb** [comment]
Where the test.sh should be placed?
I have a usecase where i  defiened a shell in a folder and azkaban job has to trigger it.
inputs needed..

**Comment by gao634209276** [comment]
@reallocf yeah, is not just a simple job, so I use azkaban resolve dependence, but I found this error when run `count1=$(more my_file | wc -l)`, I mean: I want to know the difference at this case... 
the case of killing a blocked job and still running remainning commands is difficult reappearance，
The content of the shell script is like this:
```
hdfs -dfs -getmerge /hdfs/path my_file
count1=$(more my_file | wc -l)
url='server_url'
echo "$url"
res=$(curl $url/key=$count1)
echo $res
```
